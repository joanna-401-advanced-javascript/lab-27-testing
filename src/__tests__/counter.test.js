import React from 'react';
import renderer from 'react-test-renderer';
import Counter from '../components/counter/counter';

describe('<Counter />', () => {
  it('is rendered at application start', () => {
    let app = shallow(<Counter />);
    expect(app.find('.counter').exists()).toBe(true);
    expect(app.find('.upClicker').exists()).toBe(true);
    expect(app.find('.downClicker').exists()).toBe(true);
  });

  it('is incremented when up clicker is clicked', () => {
    let app = mount(<Counter />);
    let upButton = app.find('.upClicker');
    upButton.simulate('click');
    expect(app.state('count')).toEqual(1);
  });

  it('is decremented when up clicker is clicked', () => {
    let app = mount(<Counter />);
    let downButton = app.find('.downClicker');
    downButton.simulate('click');
    downButton.simulate('click');
    expect(app.state('count')).toEqual(-2);
  });

  it('snapshot is rendered correctly', () => {
    const count = renderer
      .create(<Counter />)
      .toJSON();
    expect(count).toMatchSnapshot();
  });
});