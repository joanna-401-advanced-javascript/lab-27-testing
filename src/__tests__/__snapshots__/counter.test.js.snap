// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> snapshot is rendered correctly 1`] = `
<section
  className="counter"
>
  <a
    className="downClicker"
    href="#"
    onClick={[Function]}
  >
    -
  </a>
  <span
    className="count "
  >
    0
  </span>
  <a
    className="upClicker"
    href="#"
    onClick={[Function]}
  >
    +
  </a>
</section>
`;
